### PR TITLE
feat(nodejs): Check for `.nvmrc`

### PR DESCRIPTION
**Pull request by kidonng**
#### Description

Check for `.nvmrc` file presence in `nodejs` module.

#### Motivation and Context

Starship already checks for `.node-version`, and [`.nvmrc`](https://github.com/nvm-sh/nvm#nvmrc) is another popular alternative that worth considering.

#### How Has This Been Tested?

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

**Review by vladimyr — APPROVED** [review]
LGTM

---

Btw, it would be nice to also take `.node_version`/`.nvmrc` into account when _coloring_ node version, talking specifically about the `not_capable_style` option. But that is a matter for another PR.

**Comment by kidonng** [comment]
> Btw, it would be nice to also take `.node_version`/`.nvmrc` into account when _coloring_ node version, talking specifically about the `not_capable_style` option. But that is a matter for another PR.

That's not trivial to implement. Both file don't have a fixed format, [more info](https://github.com/tj/n#specifying-nodejs-versions).
However, if someone wants to try, `lts`, `latest` and `current` besides version numbers should be supported at least.

**Comment by vladimyr** [comment]
> That's not trivial to implement.

A smart man once told me: if it was easy I would do it myself. 😀

> Both file don't have a fixed format, making it hard to parse and handle, [more info](https://github.com/tj/n#specifying-nodejs-versions).

Jokes aside I absolutely love informed responses! You made my day ❤️ 

Here is how I see it. If it is a numeric version, complete, incomplete, w, or w/o prefix it should be supported by the existing semver parser crate we already use. Support for lts, latest and current is IMO, out of scope. Those are moving targets and supporting them requires constant changing. Argon, boron, carbon, and friends are easier to support because those names are known upfront https://github.com/nodejs/Release/blob/main/CODENAMES.md

If you think about it is a sort of progressive enhancement. You obviously can't support everything but in those cases, you don't lose anything because right now users also don't get that actionable info - like in cases having `.nvmrc` with lts inside, for example. On the other hand, we could give them valuable feedback over a subset of possibilities most importantly in cases where a concrete version is specified, being probably most widely used. That seems like a clear win to me.

**Comment by vladimyr** [comment]
Also if you agree we can create a new issue to move the convo there.

**Comment by kidonng** [comment]
> Also if you agree we can create a new issue to move the convo there.

Sure, we need a issue to track this.
